pontoon/base/models.py

```diff
@@ -284,53 +284,82 @@ def latest_activity(self, locale=None):
 
         return translations.latest_activity()
 
+    def add_detail(self, array, path, entity, translated, approved):
```

Comment:
`add_detail` is an odd method to add to the `Project` class because it doesn't actually rely on the project instance directly. I think this'd be better directly in the function, even if it's repeated twice (or maybe define an anonymous function within the other method?)